src/blockdata/script.rs

```diff
@@ -446,11 +446,9 @@ impl<'a> Iterator for Instructions<'a> {
                     self.data = &[];  // Kill iterator so that it does not return an infinite stream of errors
                     return Some(Instruction::Error(Error::EarlyEndOfScript));
                 }
-                if self.enforce_minimal {
-                    if n == 1 && (self.data[1] == 0x81 || (self.data[1] > 0 && self.data[1] <= 16)) {
-                        self.data = &[];
-                        return Some(Instruction::Error(Error::NonMinimalPush));
-                    }
+                if self.enforce_minimal && n == 1 && (self.data[1] == 0x81 || (self.data[1] > 0 && self.data[1] <= 16)) {
```

Comment:
I think in this case `clippy` is wrong, and I don't like this lint. Having a top `if self.enforce_minimal {` makes it clear, that the whole block of code applies only if `self.enforce_minimal`, without having to scan the whole expression.

I'd recommend applying an override instead to mute `clippy` here.

Reply:
This PR doesn't make the repo clippy compliant yet. we'll of course need to mute a bunch of suggestions, this is just a first try to see how people feel about this and to at least integrate the important suggestions.

I do think it's still readable because it short circuits, maybe having a new line there could help readability.

Reply:
I also think the original was more readable.

Reply:
reverted